Sources/UIExtensions/ViewModifiers/View+HorizontalMask.swift

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2022 Lautsprecher Teufel GmbH. All rights reserved.
+
+import SwiftUI
+
+extension View {
+
+    /// Adds a horizontally fade in/out effect to the view. The padding allows to start the fade effect relative to the horizontal bounds of the view.
+    public func maskHorizontally(leftPadding: CGFloat = 0, leftFade: CGFloat? = nil, rightFade: CGFloat? = nil, rightPadding: CGFloat = 0) -> some View {
```

Comment:
nitpicking: the order of parameters is a bit odd, being left padding-fade, right fade-padding, opposites. Also I guess the same technique used in .padding could be used here, like
.padding(.left, 4)
.padding(.horizontal, 5)
.padding([.left, .top], 7)

Following same logic:
.mask(.left, padding: 3, fade: 1)
.mask(.horizontal, padding: 6, fade: 2)
.mask([.left, .top], padding: 3, fade: 1)

Reply:
Also, leading and trailing instead of left and right, maybe

Reply:
About the Edges: I'm not sure how I can limit this to horizontal Edges properly. I can have a look at it again on monday